app/scripts/modules/kubernetes/container/lifecycleHook.component.ts

```diff
@@ -0,0 +1,111 @@
+import {module} from 'angular';
+
+interface IKubernetesHandler {
+  type: 'EXEC' | 'HTTP';
+  execAction?: IKubernetesExecAction;
+  httpGetAction?: IKubernetesHttpGetAction;
+}
+
+interface IKubernetesExecAction {
+  commands: string[];
+}
+
+interface IKubernetesHttpGetAction {
+  path: string;
+  port: number;
+  host: string;
+  uriScheme: string;
+  httpHeaders: IKeyValuePair[];
+}
+
+interface IKeyValuePair {
+  name: string;
+  value: string;
+}
+
+class KubernetesLifecycleHookConfigurerCtrl implements ng.IComponentController {
+  public heading: string;
+  public handler: IKubernetesHandler;
+  public execActionCommandsViewValue: string;
+  private onHandlerChange: Function;
+  private handlerEnabled = false;
+
+  public $onInit(): void {
+    this.handlerEnabled = !!this.handler;
+    if (this.handlerEnabled) {
+      this.initializeHandler();
+    }
+  }
+
+  public toggleHandlerEnabled(): void {
+    if (this.handlerEnabled) {
+      this.initializeHandler();
+    } else {
+      this.handler = null;
+      this.execActionCommandsViewValue = null;
+      this.onHandlerChange({handler: null});
+    }
+  }
+
+  public handlerTypeModelToViewValue(handlerTypeModelValue: 'EXEC' | 'HTTP'): 'exec' | 'httpGet' {
```

Comment:
If this is client-side only, I think it's totally ok to use an `enum`.  We haven't talked about an official policy, however.  It reads a lot nicer here and elsewhere in the code.